pkg/infra/kubernetes/exec_unit.go

```diff
@@ -167,7 +177,91 @@ func (unit *HelmExecUnit) transformDeployment(cfg config.ExecutionUnit) ([]HelmC
 	return values, nil
 }
 
-func (unit *HelmExecUnit) transformService() (values []HelmChartValue, err error) {
+func (unit *HelmExecUnit) transformHorizontalPodAutoscaler(cfg config.ExecutionUnit) ([]HelmChartValue, error) {
+	log := zap.L().Sugar().With(logging.FileField(unit.HorizontalPodAutoscaler), zap.String("unit", unit.Name))
+	log.Debugf("Transforming file, %s, for exec unit, %s", unit.HorizontalPodAutoscaler.Path(), unit.Name)
+	obj, err := readFile(unit.HorizontalPodAutoscaler)
+	if err != nil {
+		return nil, nil
+	}
+	hpa, ok := obj.(*autoscaling.HorizontalPodAutoscaler)
+	if !ok {
+		err = fmt.Errorf("expected file %s to contain HorizontalPodAutoscaler Kind", unit.HorizontalPodAutoscaler.Path())
+		return nil, nil
+	}
+	k8Cfg := cfg.GetExecutionUnitParamsAsKubernetes()
+	hpaCfg := k8Cfg.HorizontalPodAutoScalingConfig
+
+	if k8Cfg.Replicas != 0 {
+		minReplicas := int32(k8Cfg.Replicas)
```

Comment:
im not sure we want to set this to what they specify the replicas add because i could see scenarios in which they would want to scale down. Not blocking though since its our current UX